### PR TITLE
Bugfix for badly formatted date time string

### PR DESCRIPTION
**Pull request by annando**
There was one place where we hadn't formatted a date time value according to the SQL standard. To make that easier, I added a new function. By now we are (mis)using "datetime_convert" for that. This is a rather huge function that is oversized for this functionality.

### CODE REVIEW

File: include/dbm.php [hunk 1]
```diff
@@ -86,5 +86,16 @@ private static function esc_array_callback(&$value, $key, $add_quotation) {
...
+
+	/**
+	 * Checks Converts any date string into a SQL compatible date string
+	 *
+	 * @param string $date a date string in any format
+	 * @return string SQL style date string
+	 */
+	public static function date($date) {
+		$timestamp = strtotime($date);
+		return date('Y-m-d H:i:s');
```
Comment (Hypolite) [inline include/dbm.php]: This function should return the current timestamp by default, and `$timestamp` is never used.

File: include/socgraph.php [hunk 2]
```diff
@@ -622,7 +622,7 @@ function poco_last_updated($profile, $force = false) {
 			$last_updated = "0000-00-00 00:00:00";
 
 	q("UPDATE `gcontact` SET `updated` = '%s', `last_contact` = '%s' WHERE `nurl` = '%s'",
-		dbesc($last_updated), dbesc(datetime_convert()), dbesc(normalise_link($profile)));
+		dbesc(dbm::date($last_updated)), dbesc(datetime_convert()), dbesc(normalise_link($profile)));
```
Comment (Hypolite) [inline include/socgraph.php]: Once you're done with my first comment, you can replace `datetime_convert()` by `dbm::date()`.

File: include/dbm.php (hunk 1 above)
Comment (annando) [inline include/dbm.php]: Örgs ...

Thanks for the hint.

**Comment by annando** [comment]
Sorry for another change ;-) I just had double checked if the lowest value could make problems.

**Pull request merged**